### PR TITLE
new command (1st try) - _RGB_CONVERT_TO_SF

### PR DESCRIPTION
**Pull request by RemCaren**

### CODE REVIEW

**Comment by dgirardeau** [comment]
You miss the command registration part ;)

File: qCC/ccCommandLineCommands.cpp [hunk 1]
```diff
@@ -1842,6 +1843,50 @@ bool CommandSFConvertToRGB::process(ccCommandLineInterface &cmd)
...
+
+		if (!cloud->hasColors())
+		{
+			cmd.warning(QObject::tr("Cloud %1 has no colors").arg(cloud->getName()));
+			continue;
+		}
+
+		ccHObject::Container container;
+		container.push_back(cloud);
+		if (!ccEntityAction::sfFromColors(container, cmd.widgetParent()))
```
Comment (dgirardeau) [inline qCC/ccCommandLineCommands.cpp:1870]: you should normally call the other sfFromColors method (the one with booleans as input parameters). For now you can set them all to 'true', but you may also want to add options to the command so that users can specify which component they want to export).

**Review by dgirardeau — CHANGES_REQUESTED** [review]

**Comment by RemCaren** [comment]
Isn’t this method directly called by the other ?

De : Daniel Girardeau-Montaut
Envoyé le :lundi 8 novembre 2021 11:44
À : CloudCompare/CloudCompare
Cc : RemCaren; Author
Objet :Re: [CloudCompare/CloudCompare] new command (1st try) -_RGB_CONVERT_TO_SF (PR #1538)

@dgirardeau requested changes on this pull request.

In qCC/ccCommandLineCommands.cpp:
> +		return cmd.error(QObject::tr("No point cloud on which to convert RGB to SF! (be sure to open one with \"-%1 [cloud filename]\" before \"-%2\")").arg(COMMAND_OPEN, COMMAND_RGB_CONVERT_TO_SF));
+	}
+
+	for (CLCloudDesc& thisCloudDesc : cmd.clouds())
+	{
+		ccPointCloud* cloud = thisCloudDesc.pc;
+
+		if (!cloud->hasColors())
+		{
+			cmd.warning(QObject::tr("Cloud %1 has no colors").arg(cloud->getName()));
+			continue;
+		}
+
+		ccHObject::Container container;
+		container.push_back(cloud);
+		if (!ccEntityAction::sfFromColors(container, cmd.widgetParent()))
you should normally call the other sfFromColors method (the one with booleans as input parameters). For now you can set them all to 'true', but you may also want to add options to the command so that users can specify which component they want to export).
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by dgirardeau** [comment]
It is, but it will display a dialog to ask the user to select the components. And we don't want any dialog to popup in the command line mode ;)

**Comment by RemCaren** [comment]
Could it work with a cmd.arguments() for each bool ?

De : Daniel Girardeau-Montaut
Envoyé le :lundi 8 novembre 2021 11:49
À : CloudCompare/CloudCompare
Cc : RemCaren; Author
Objet :Re: [CloudCompare/CloudCompare] new command (1st try) -_RGB_CONVERT_TO_SF (PR #1538)

It is, but it will display a dialog to ask the user to select the components. And we don't want any dialog to popup in the command line mode ;)
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by RemCaren** [comment]
A little bit like you did for the sf_to_rgb method

De : Daniel Girardeau-Montaut
Envoyé le :lundi 8 novembre 2021 11:49
À : CloudCompare/CloudCompare
Cc : RemCaren; Author
Objet :Re: [CloudCompare/CloudCompare] new command (1st try) -_RGB_CONVERT_TO_SF (PR #1538)

It is, but it will display a dialog to ask the user to select the components. And we don't want any dialog to popup in the command line mode ;)
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by dgirardeau** [comment]
When you reply to the emails instead of github, it makes things a little bit hard to follow 😅 

You could add an argument (optional or not) as a string that can be any sequence of 'R', 'G', 'B', 'A' and 'C' (composite). And then you deduce which component is active by checking if the string contains each of these letters? (with QString::contains)

**Comment by RemCaren** [comment]
Sry i'll write on github though...
I see what you mean, but how do you ask the user a str ? I

**Comment by RemCaren** [comment]
And yeah i made a mistake by making 2 PRs, sorry...

**Comment by RemCaren** [comment]
Otherwise, couldn't we put all of these on 'true' (or just the composite for example) for a 1st version, which could be updated later ? I would need it for this afternoon if possible, so i'm a bit in a hurry :o

**Comment by dgirardeau** [comment]
Yes, no problem for a default version all 'trues' for now.
